chain/chain/src/store.rs

```diff
@@ -348,6 +348,10 @@ pub struct ChainStore {
     block_ordinal_to_hash: LruCache<Vec<u8>, CryptoHash>,
     /// Processed block heights.
     processed_block_heights: LruCache<Vec<u8>, ()>,
+    /// Is this an archival node?
+    /// TODO: this only exists because archival nodes don't need ColTrieChanges.
+    /// Should remove this field if and when that whole column is deleted.
```

Comment:
I think it's possible and maybe desirable to get rid of it for non-archival nodes. Im referring to the idea in this comment: https://github.com/near/nearcore/issues/6250#issuecomment-1031409797

although even given that I could see an argument for not including the TODO if theres no tracking issue/real plans to get rid of it. WDYT?

Reply:
hmm will it? I guess i'm not sure how it would end up being implemented, but I was thinking that the logic to replace the current `ColTrieChanges` would still just be encapsulated in `ChainStoreUpdate::clear_block_data()`, and that function is already called conditionally based on whether it's archival because the `Client` has that info:

```rust
            if !self.config.archive {
                let timer = metrics::GC_TIME.start_timer();
                if let Err(err) = self
                    .chain
                    .clear_data(self.runtime_adapter.get_tries(), self.config.gc_blocks_limit)
                {
...
```

The difference is now just that `clear_data()` does something different than reading from `ColTrieChanges` to find out what to do

Reply:
wait but isnt the question here whether we want to  delete the column in RPC nodes, too? As in the comment I linked above. If that's not actually a good idea/the right path forward then maybe we just delete the TODO here, yeah